internal/pkg/agent/application/coordinator/diagnostics_test.go

```diff
@@ -0,0 +1,397 @@
+						t.Logf("Substituting host id %q with %q", hostMap["id"], hostIDPlaceholder)
+						hostMap["id"] = hostIDPlaceholder
+					}
+				}
+			}
+
+			if assert.Containsf(t, yamlContent, pathKey, "config yaml does not contain agent path map") {
+				pathValue := yamlContent[pathKey]
+				if assert.IsType(t, map[interface{}]interface{}{}, pathValue) {
+					pathMap := pathValue.(map[interface{}]interface{})
+					currentDir := pathMap["config"].(string)
+					for _, key := range []string{"config", "data", "home", "logs"} {
+						if assert.Containsf(t, pathMap, key, "path map is missing expected key %q", key) {
+							value := pathMap[key]
+							if assert.IsType(t, "", value) {
+								valueString := value.(string)
+								valueString = strings.Replace(valueString, currentDir, agentPathPlaceholder, 1)
+								pathMap[key] = filepath.ToSlash(valueString)
+							}
+						}
+					}
+				}
+			}
+		} else if fileName == "components.yaml" {
+			rawComponents, ok := yamlContent["components"].([]any)
+
+			if assert.True(t, ok, "unexpected component format in file %s", fileName) {
+				sort.Sort(SortByID(rawComponents))
+				// fix the paths to forward slash for each
+				for _, comp := range rawComponents {
+					compInputSpec := comp.(map[any]any)["input_spec"].(map[any]any)
+					compInputSpec["binary_path"] = filepath.ToSlash(compInputSpec["binary_path"].(string))
+				}
+				yamlContent["components"] = rawComponents
+			}
+
+		}
+		sanitizedBytes, err := yaml.Marshal(yamlContent)
+		assert.NoError(t, err)
+		return string(sanitizedBytes)
+	}
+
+	//substitute current running dir with a placeholder
+	testDir := path.Dir(os.Args[0])
+	t.Logf("Replacing test dir %s with %s", testDir, agentPathPlaceholder)
+	return strings.ReplaceAll(string(rawBytes), testDir, agentPathPlaceholder)
+}
+
+type SortByID []any
+
+func (a SortByID) Len() int      { return len(a) }
+func (a SortByID) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
+func (a SortByID) Less(i, j int) bool {
+	// ugly but ids are mandatory and they must be a string, panic otherwise
+	iID := a[i].(map[any]any)["id"].(string)
+	jID := a[j].(map[any]any)["id"].(string)
+	return iID < jID
+}
+
+type coordinatorTestHelper struct {
+	coordinator *coordinator.Coordinator
+
+	runtimeManager      *mocks.RuntimeManager
+	runtimeErrorChannel chan error
+
+	configManager       *mocks.ConfigManager
+	configChangeChannel chan coordinator.ConfigChange
+	configErrorChannel  chan error
+	actionErrorChannel  chan error
+
+	varsManager      *mocks.VarsManager
+	varsChannel      chan []*transpiler.Vars
+	varsErrorChannel chan error
+
+	capability     *mocks.Capability
+	upgradeManager *mocks.UpgradeManager
+	reExecManager  *mocks.ReExecManager
+	monitorManager *mocks.MonitorManager
+}
+
+func newCoordinatorTestHelper(t *testing.T, agentInfo *info.AgentInfo, specs component.RuntimeSpecs, isManaged bool) *coordinatorTestHelper {
+	helper := new(coordinatorTestHelper)
```

Comment:
[Suggestion]
mark it as a helper function:
```suggestion
func newCoordinatorTestHelper(t *testing.T, agentInfo *info.AgentInfo, specs component.RuntimeSpecs, isManaged bool) *coordinatorTestHelper {
	t.Helper()
	helper := new(coordinatorTestHelper)
```